applications/base/skills.py

```
@@ -534,21 +1008,269 @@ class industry_test_implementation(BaseTool):
+    camera_name: Optional[str] = Field(default='camera', description="Camera to be used to compare the images")
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+class compare_images_implementation(BaseTool):
+    """Compare two images and return the difference between them based on the given prompt."""
+    name: ClassVar[str] = "compare_images"
+    description: ClassVar[str] = "Tool to compare two images and return the difference between them based on the given prompt"
+    args_schema: ClassVar[Type[BaseModel]] = compare_images_definition
+    return_direct: ClassVar[bool] = False
+    verbose: ClassVar[bool] = False
+
+    def _run(self, prompt: str = None, camera_name: str = "camera") -> str:
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)